Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp

```diff
@@ -186,19 +186,50 @@ namespace O3DE::ProjectManager
                     // First try <project-build-path>/bin/$<CONFIG> and if that path doesn't exist
                     // try <project-build-path>/bin/$<PLATFORM>/$<CONFIG>
                     buildConfigurationPath /= "bin";
-                    if (editorPath = (buildConfigurationPath / AZ_BUILD_CONFIGURATION_TYPE / "Editor").
+                    auto paths = AZStd::vector<AZ::IO::FixedMaxPath>{
+                        buildConfigurationPath / AZ_BUILD_CONFIGURATION_TYPE / "Editor",
+                        buildConfigurationPath / AZ_TRAIT_OS_PLATFORM_CODENAME / AZ_BUILD_CONFIGURATION_TYPE / "Editor"
+                    };
+
+                    // always try profile config because that is the default
+                    if (strcmp(AZ_BUILD_CONFIGURATION_TYPE, "profile") != 0)
+                    {
+                        paths.emplace_back(buildConfigurationPath / "profile" / "Editor");
+                        paths.emplace_back(buildConfigurationPath / AZ_TRAIT_OS_PLATFORM_CODENAME / "profile" / "Editor");
+                    }
+
+                    for (auto path : paths)
+                    {
+                        if(AZ::IO::SystemFile::Exists(path.ReplaceExtension(AZ_TRAIT_OS_EXECUTABLE_EXTENSION).c_str()))
+                        {
+                            return path;
+                        }
+                    }
+                }
+            }
+
+            // Attempt to find the Editor in the specified engine's bin folder
+            if(auto engineResult = PythonBindingsInterface::Get()->GetProjectEngine(projectPath.Native().data()); engineResult)
+            {
+                auto engineInfo = engineResult.GetValue<EngineInfo>();
+                if (!engineInfo.m_thisEngine)
+                {
+                    AZ::IO::FixedMaxPath fixedEnginePath{ engineInfo.m_path.toStdString().c_str() };
+                    // first try the default sdk path
+                    if (editorPath = (fixedEnginePath / "bin" / AZ_TRAIT_OS_PLATFORM_CODENAME / "profile" / "Default" / "Editor").
                         ReplaceExtension(AZ_TRAIT_OS_EXECUTABLE_EXTENSION);
                         AZ::IO::SystemFile::Exists(editorPath.c_str()))
                     {
                         return editorPath;
                     }
-                    else if (editorPath = (buildConfigurationPath / AZ_TRAIT_OS_PLATFORM_CODENAME
-                        / AZ_BUILD_CONFIGURATION_TYPE / "Editor").
+                    // next try the default build path
+                    else if (editorPath = (fixedEnginePath / "build" / "windows_vs2019" / "bin" / "profile" / "Editor").
```

Comment:
We shouldn't hardcode "windows_vs2019" into the path check.

This should be queried, especially since the build directory that is used on Windows by **ProjectManager** is hardcoded to "[windows](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Platform/Windows/ProjectManagerDefs_windows.cpp#L12)" not "windows_vs2019"

Reply:
Also the implementation for this code search logic on Linux and and Mac is missing.

Reply:
My thinking here was, if a user followed our [default build the engine documentation](https://www.o3de.org/docs/welcome-guide/setup/setup-from-github/building-windows/#build-the-engine) which uses "build/windows_vs2019", then the binaries might be there.  This directory is also used in the [documentation for running the editor for a source engine](https://www.o3de.org/docs/welcome-guide/create/creating-projects-using-cli/creating-windows/#build-the-o3de-project).  Thoughts?

Reply:
The other tricky one is if a user creates their own SDK and distributes it to their team with some build path other than the default, how will Project Manager find the SDK engine binaries on a user's machine where they haven't compiled the project or the engine?  It seems in this case we'd need an entry in the `engine.json` specifying where the relative path to the binaries.

Reply:
We don't need an entry in the `engine.json`, when CMake configures, we generate a Settings [Registry file with the path relative path information to the build directory](https://github.com/o3de/o3de/blob/development/cmake/Projects.cmake#L95-L122) in it when configuring for monolithic builds or on a platform where we don't build any tools application(Android and iOS).

We can just update that logic to always generate by removing the monolithic build checks

Reply:
It looks like that `project_build_path` setreg key will provide the project binary path, not the engine binary path where the Editor.exe is.

Also, it appears that on Mac we write a `AZ::SettingsRegistryMergeUtils::FilePathKey_InstalledBinaryFolder` key which contains the path to the engine binaries - maybe we could do something similar for the other platforms?

Reply:
Ah it looks like the version of OSX I have (Catalina) is no longer supported by O3DE le-sigh.

Reply:
Yes we can write out the relative path for the Editor for other platforms into a [BinariesInstallPath.setreg](https://github.com/o3de/o3de/blob/development/cmake/Platform/Mac/Install_mac.cmake#L11-L36) file.

But we shouldn't use the exact logic that is posted in the Install_Mac.cmake script in the install scripts for Linux and Windows.
That logic is using the `CMAKE_RUNTIME_OUTPUT_DIRECTORY` for the binaries directory, where the directory we actually want is the directory where Editor gets output to, which may be a subdirectory of the `CMAKE_RUNTIME_OUTPUT_DIRECTORY`

Also FYI this is the value of @runtime_output_directory@ variable that is being configured into the BinariesInstallPath.setreg: https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Install_common.cmake#L50

Reply:
What I think we should do is update the template in the [Install_mac.cmake](https://github.com/o3de/o3de/blob/development/cmake/Platform/Mac/Install_mac.cmake#L15-L27) script add a couple of new SettingsRegistry keys. `InstallEditorFolder` and `InstalledAssetProcessorFolder`.

We can look up the RUNTIME_OUTPUT_DIRECTORY properties of the Editor and AssetProcessor TARGETS in cmake to find their exact directory.
```cmake
get_target_property(editor_runtime_output_directory Editor RUNTIME_OUTPUT_DIRECTORY)
get_target_property(assetprocessor_runtime_output_directory AssetProcessor RUNTIME_OUTPUT_DIRECTORY)
```

Putting it all together we can update the `Install_<platform>.cmake` scripts to have the following
```cmake
set(installed_binaries_path_template [[
{
    "Amazon": {
        "AzCore": {
            "Runtime": {
                "FilePaths": {
                    "InstalledBinariesFolder": "@runtime_output_directory@",
                    "InstalledEditorFolder": "@editor_runtime_output_directory@",
                    "InstalledAssetProcessorFolder": "@assetprocessor_runtime_output_directory@"
                }
            }
        }
    }
}]]
)

get_target_property(editor_runtime_output_directory Editor RUNTIME_OUTPUT_DIRECTORY)
get_target_property(assetprocessor_runtime_output_directory AssetProcessor RUNTIME_OUTPUT_DIRECTORY)

# This setreg file will be used by all of our installed app bundles to locate installed
# runtime dependencies. It contains the path to binary install directory relative to 
# the installed engine root.
string(CONFIGURE "${installed_binaries_path_template}" configured_setreg_file)
file(GENERATE
    OUTPUT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/BinariesInstallPath.setreg
    CONTENT "${configured_setreg_file}"
)
```

Reply:
In order to make sure we at least get support for Windows & Linux in before we cut the next release branch, I'm going to break out the Mac support into a separate ticket https://github.com/o3de/o3de/issues/10971 and reference these comments as the intended solution.

Reply:
The next release branch is being cut in 4 weeks.

The code I posted would solve the issue for all platforms.
It just needs to be put in the appropriate `Install_mac.cmake`, `Install_linux.cmake` and `Install_windows.cmake` files.